datalad/support/gitrepo.py

```diff
@@ -2362,15 +2370,115 @@ def gc(self, allow_background=False, auto=False):
             cmd_options += ['--auto']
         self._git_custom_command('', cmd_options)
 
-    def get_submodules(self, sorted_=True):
-        """Return a list of git.Submodule instances for all submodules"""
-        # check whether we have anything in the repo. if not go home early
-        if not self.repo.head.is_valid():
-            return []
-        submodules = self.repo.submodules
+    def _parse_gitmodules(self):
+        # TODO read .gitconfig from Git blob?
+        gitmodules = self.pathobj / '.gitmodules'
+        if not gitmodules.exists():
+            return {}
+        # pull out file content
+        out, err = self._git_custom_command(
+            '',
+            ['git', 'config', '-z', '-l', '--file', '.gitmodules'])
+        # abuse our config parser
+        db, _ = _parse_gitconfig_dump(out, {}, None, True)
+        mods = {}
+        for k, v in iteritems(db):
+            if not k.startswith('submodule.'):
+                # we don't know what this is
+                lgr.debug("Skip unrecognized .gitmodule specification: %s=%s", k, v)
```

Comment:
I wonder if, while at it, we should raise to to WARNING level?!  Hiding possible problems and unexpected situations could just cost us in the long run